### PR TITLE
patches: improve ctypes patch for python 3.5

### PR DESCRIPTION
**Pull request by sergiusens**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

**Comment by sergiusens** [comment]
A potential error in the python plugin seen on other PRs is solved by #2077

**Comment by sergiusens** [comment]
And the only error is fixed by the aforementioned PR.

File: tests/integration/general/test_elf.py [hunk 1]
```diff
@@ -106,8 +106,9 @@ def _setup_project(self, keep_execstack: bool):
         snapcraft_yaml.update_part('test-part', {
             'plugin': 'nil',
             'build-attributes': attributes,
-            'build': ('/usr/sbin/execstack --set-execstack '
-                      '$SNAPCRAFT_PART_INSTALL/usr/bin/hello'),
+            'build': (
+                'execstack --set-execstack '
+                '$SNAPCRAFT_PART_INSTALL/usr/bin/hello'),
```
Comment (zyga) [inline tests/integration/general/test_elf.py]: Is this a pure whitespace change?

File: patches/ctypes_init.diff [hunk 2]
```diff
@@ -14,11 +13,36 @@
...
+ class CDLL(object):
+     """An instance of this class represents a loaded dll/shared
+     library, exporting functions using the standard C calling
+@@ -344,7 +354,15 @@
+         self._FuncPtr = _FuncPtr
+ 
+         if handle is None:
+-            self._handle = _dlopen(self._name, mode)
++            name = self._name
++            if name is not None and _os.getenv('SNAP_NAME', '') == 'snapcraft':
```
Comment (zyga) [inline patches/ctypes_init.diff]: It would be good to document why this indirection is needed

Comment (sergiusens) [reply patches/ctypes_init.diff]: ah, that is legacy, but by default cpython loads only what is in ld cache, so it wouldn't pick up any library inside the snap, bad thing for classic :wink:

**Pull request merged**